packages/next/src/client/link.tsx

```diff
@@ -280,6 +280,36 @@ function formatStringOrUrl(urlObjOrString: UrlObject | string): string {
   return formatUrl(urlObjOrString)
 }
 
+/**
+ * Returns the ref of a React element handling differences between React 19 and older versions.
+ * It will throw runtime error if the element is not a valid React element.
+ *
+ * The source is a copy and paste of https://github.com/radix-ui/primitives/blob/6e75e117977c9e6ffa939e6951a707f16ba0f95e/packages/react/presence/src/presence.tsx#L173
+ * It's also duplicated in the App Router source.
+ * @param element React.ReactElement
+ * @returns React.Ref<any> | undefined
+ */
+function getReactElementRef(
+  element: React.ReactElement<{ ref?: React.Ref<unknown> }>
+): React.Ref<any> | undefined {
+  // React <=18 in DEV
```

Comment:
Can we just shortcut to the deprecated version in prod? Renders are a hotpath so we should be as nimble as possible. The goal here is to avoid the dev-only warning, no?